.github/workflows/ci.yml

```diff
@@ -75,21 +75,39 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        artifact: [helloworld, raqote-winit]
+        artifact: [helloworld, raqote-winit, template]
     runs-on: ubuntu-latest
     needs: build-ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+
+    # install x
     - uses: actions/download-artifact@v2
```

Comment:
@dvc94ch Can we just use `name: Install X` etc for these instead of comments, so that they show up nicely in the final graph too? Right now it's all just `Run <action name>` (and `Run <the command>` for non-`uses` things: the shell scripts).

Reply:
The comments group multiple steps into objectives. I don't think giving each step a name is useful, and probably to verbose

Reply:
there is a feature called composite action which seems like it could help reduce a lot of duplication:
https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

Reply:
> The comments group multiple steps into objectives. I don't think giving each step a name is useful, and probably to verbose

Fair enough, it might still be nice to group some of them.



> there is a feature called composite action which seems like it could help reduce a lot of duplication:
> https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

Heh that one fell off my todo list, wanted to try it out for another project too. Xbuild can definitely benefit with lots of copypasted code. Who should look into it? I haven't kept up with xbuild last week but slowly getting back into it.

Reply:
not sure if messing about with composite actions is the best use of my time right now

Reply:
I'll try to look into it when in need of something light.

Reply:
If you want to look into it that's great. Not sure if it's the most important thing for xbuild right now

Reply:
sorry for deleting comments, there seem to be some races causing a bad ordering of messages :)

Reply:
Having good, functioning CI is important in my eyes. Having it maintainable and concise doubly so, right now it's pretty nontrivial to alter, replicate, or debug individual jobs/steps due to the vast number of lines.

Reply:
> sorry for deleting comments, there seem to be some races causing a bad ordering of messages :)

I had only noticed that there wasn't a message when clicking the notification and replying to what I spotted in there, no worries :D